src/foam/nanos/column/NestedPropertiesExpression.js

```diff
@@ -81,61 +81,95 @@ foam.CLASS({
+        else thisPropExpr = prop;
+
+        if ( ! expr ) expr = thisPropExpr;
+        else foam.mlang.Expressions.create().DOT(expr, thisPropExpr);
+
+        if ( i === propName.length - 1 )
+          return expr;
+
+        return this.returnDotExprForNestedProperty(prop.of, propName, ++i, expr);
       },
       javaCode: `
         ClassInfo ci = of;
-        PropertyInfo p = (PropertyInfo) ci.getAxiomByName(propName[i]);
+        PropertyInfo prop = (PropertyInfo) ci.getAxiomByName(propName[i]);
+
+        if ( prop == null ) return null; 
+
+        Boolean isPropAReference = isPropertyAReference(ci, prop);
+
+        Expr thisPropExpr = null;
+        if ( isPropAReference )
+          thisPropExpr = REF(prop);
+        else
+          thisPropExpr = prop;
+        if ( expr == null) expr = thisPropExpr;
+        else expr = DOT(expr, thisPropExpr);
+
         if ( i == propName.length - 1 ) {
-          return p;
+          return expr;
         }
         
-        StringBuilder sb = new StringBuilder("find");
         try {
+          StringBuilder sb = new StringBuilder("find");
           Class cls;
-          if ( p instanceof foam.core.AbstractFObjectPropertyInfo ) {
-            cls = p.getValueClass();
+          if ( prop instanceof foam.core.AbstractFObjectPropertyInfo ) {
+            cls = prop.getValueClass();
           } else {
             sb.setLength(4);
-            Method m = ci.getObjClass().getMethod(StringUtil.capitalize(p.getName()), foam.core.X.class);
+            sb.append(StringUtil.capitalize(prop.getName()));
+            Method m = ci.getObjClass().getMethod(sb.toString(), foam.core.X.class);
             cls = m.getReturnType();
           }
           ci = (ClassInfo) cls.getMethod("getOwnClassInfo").invoke(null);
-        } catch (Exception e) {
-          Logger logger = (Logger) getX().get("logger");
-          logger.error(e);
+        } catch( Throwable t ) {
           return null;
         }
-
-        return returnPropExpr(p, returnDotExprForNestedProperty(ci, propName, ++i));
+        
+        return returnDotExprForNestedProperty(ci, propName, ++i, expr);
       `
     },
     {
-      name: 'returnPropExpr',
-      javaType: 'foam.mlang.Expr',
+      name: 'isPropertyAReference',
+      javaType: 'Boolean',
       args: [
         {
-          name: 'prop1',
-          class: 'Object',
-          javaType: 'foam.mlang.Expr'
+          name: 'ci',
+          type: 'Object',
+          javaType: 'foam.core.ClassInfo'
         },
         {
-          name: 'prop2',
-          class: 'Object',
-          javaType: 'foam.mlang.Expr'
-        }
+          name: 'prop',
+          javaType: 'foam.core.PropertyInfo',
+          javaInfoType: 'foam.core.AbstractObjectPropertyInfo'
+        },
       ],
-      code: function (prop1, prop2) {
-        return foam.mlang.Expressions.create().DOT(prop1, prop2);
+      code: function (of, propName, i, expr) {
+        return false;
```

Comment:
is it always false for js?

Reply:
it was kinda not implemented.  sorry about that I've cleaned it up.